### PR TITLE
Controllers

### PR DESCRIPTION
**Pull request by MarianMilian**

### CODE REVIEW

File: src/main/java/team2/spring/library/services/BookServiceImpl.java [hunk 1]
```diff
@@ -28,6 +29,7 @@ public BookDto isBookAvailable(BookDto bookDto) {
 
   @Override
   public List<Book> findBooksByAuthor(Author author) {
-    return bookDaoInfs.findBooksByAuthor(author);
+    Author author1 = authorDaoInfs.findByName(author.getName());
```
Comment (DmytroDovhal) [inline src/main/java/team2/spring/library/services/BookServiceImpl.java]: author1 its bad name of obj

File: src/main/java/team2/spring/library/controllers/BookController.java [hunk 2]
```diff
@@ -96,4 +96,15 @@ public String addBook(@ModelAttribute("book") Book book, Model model) {
...
   }
+
+  @GetMapping("/getPopularBookForm")
+  public String getPopularBookForm(Model model){
+    model.addAttribute("bookDto",new BookDto());
+      return "/booksJsp/getPopularBookForm";
+  }
+  @PostMapping("/getPopularBook")
+  public String getPopularBook(@ModelAttribute("book")Book book,Model model){
+//    model.addAttribute("",);
```
Comment (RomanZahorui) [inline src/main/java/team2/spring/library/controllers/BookController.java]: remove commented

File: src/main/java/team2/spring/library/controllers/ReaderController.java [hunk 3]
```diff
@@ -23,18 +23,29 @@
   @GetMapping("/getBlackList")
   public String getBlackList(Model model) {
     model.addAttribute("readerBlackList", readerService.getBlackList());
-    return "getBlackList";
+    return "/readersJsp/getBList";
```
Comment (RomanZahorui) [inline src/main/java/team2/spring/library/controllers/ReaderController.java]: blist ??

**Review by RomanZahorui — CHANGES_REQUESTED** [review]

**Review by MarianMilian — COMMENTED** [review]
done

File: src/main/java/team2/spring/library/controllers/ReaderController.java (hunk 3 above)
Comment (MarianMilian) [reply src/main/java/team2/spring/library/controllers/ReaderController.java]: fixed

**Review by DmytroDovhal — APPROVED** [review]

**Pull request merged**